pkg/wallet/account/account.go

```diff
@@ -0,0 +1,246 @@
+}
+
+// NewFromPrivateKey creates a new account from a private key. It destroys the password.
+func NewFromPrivateKey(password *memguard.LockedBuffer, nickname string, privateKeyText *memguard.LockedBuffer) (*Account, error) {
+	version := uint8(AccountLastVersion)
+
+	var salt [16]byte
+	_, err := rand.Read(salt[:])
+	if err != nil {
+		return nil, fmt.Errorf("generating random salt: %w", err)
+	}
+
+	var nonce [12]byte
+	_, err = rand.Read(nonce[:])
+	if err != nil {
+		return nil, fmt.Errorf("generating random nonce: %w", err)
+	}
+
+	seed, privateKeyVersion, err := base58.CheckDecode(string(privateKeyText.Bytes()[1:])) // omit the first byte because it's 'S' for secret key
+	seedBuffer := memguard.NewBufferFromBytes(seed)
+	if err != nil {
+		seedBuffer.Destroy()
+		return nil, fmt.Errorf("%w: decoding base58 private key: %w", ErrInvalidPrivateKey, err)
+	}
+
+	privateKey := memguard.NewBufferFromBytes(ed25519.NewKeyFromSeed(seedBuffer.Bytes()))
+	seedBuffer.Destroy()
+
+	encryptedSecret, err := seal(privateKey, password, salt[:], nonce[:])
+	if err != nil {
+		return nil, fmt.Errorf("sealing secret: %w", err)
+	}
+
+	encryptedPrivateKey := types.EncryptedPrivateKey{
+		Object: &object.Object{
+			Kind:    object.EncryptedPrivateKey,
+			Version: privateKeyVersion,
+			Data:    encryptedSecret,
+		},
+	}
+
+	err = encryptedPrivateKey.Validate(privateKeyVersion, object.EncryptedPrivateKey)
+	if err != nil {
+		return nil, fmt.Errorf("validating encrypted private key: %w", err)
+	}
+
+	publicKey, err := encryptedPrivateKey.PublicKey(password, salt[:], nonce[:])
+	password.Destroy()
+	if err != nil {
+		return nil, fmt.Errorf("getting public key from encrypted private key: %w", err)
+	}
+
+	address := types.NewAddressFromPublicKey(publicKey)
+
+	return New(version, nickname, address, salt, nonce, &encryptedPrivateKey, publicKey)
+}
+
+// seal encrypts the private key with the password.
+func seal(privateKey, password *memguard.LockedBuffer, salt, nonce []byte) ([]byte, error) {
+	aeadCipher, secretKey, err := crypto.NewSecretCipher(password.Bytes(), salt[:])
+	if err != nil {
+		return nil, fmt.Errorf("creating secret cipher: %w", err)
+	}
+	encryptedSecret := crypto.SealSecret(aeadCipher, nonce[:], privateKey)
+
+	secretKey.Destroy()
+
+	return encryptedSecret, nil
+}
+
+// PrivateKeyTextInClear returns the private key in clear and destroys the password.
+func (a *Account) PrivateKeyTextInClear(password *memguard.LockedBuffer) (*memguard.LockedBuffer, error) {
+	return a.CipheredData.PrivateKeyTextInClear(password, a.Salt[:], a.Nonce[:], a.CipheredData.Data)
+}
+
+// Sign signs the data with the private key and destroys the password.
+func (a *Account) Sign(password *memguard.LockedBuffer, data []byte) ([]byte, error) {
+	return a.CipheredData.Sign(password, a.Salt[:], a.Nonce[:], data)
+}
+
+func (a *Account) Marshal() ([]byte, error) {
+	yamlMarshaled, err := yaml.Marshal(a)
```

Comment:
why don't you directly return yaml.Marshal(a) ? Except if you want to wrap it using  sentinel error, but you are not doing so...